c10/core/SymIntNodeImpl.h

```diff
@@ -27,60 +27,115 @@ class C10_API SymIntNodeImpl : public c10::intrusive_ptr_target {
   virtual SymIntNode add(const SymIntNode& other) {
     TORCH_CHECK(false, "NYI");
   };
+  virtual SymFloatNode add(const SymFloatNode& other) {
+    TORCH_CHECK(false, "NYI");
+  };
   virtual SymIntNode sub(const SymIntNode& other) {
     TORCH_CHECK(false, "NYI");
   };
+  virtual SymFloatNode sub(const SymFloatNode& other) {
+    TORCH_CHECK(false, "NYI");
+  };
   virtual SymIntNode mul(const SymIntNode& other) {
     TORCH_CHECK(false, "NYI");
   };
+  virtual SymFloatNode mul(const SymFloatNode& other) {
+    TORCH_CHECK(false, "NYI");
+  };
   virtual SymFloatNode truediv(const SymIntNode& other) {
     TORCH_CHECK(false, "NYI");
   };
+  virtual SymFloatNode truediv(const SymFloatNode& other) {
+    TORCH_CHECK(false, "NYI");
+  };
   virtual SymIntNode floordiv(const SymIntNode& other) {
     TORCH_CHECK(false, "NYI");
   };
-  virtual SymIntNode mod(const SymIntNode& other) {
+  virtual SymIntNode floordiv(const SymFloatNode& other) {
     TORCH_CHECK(false, "NYI");
   };
+  virtual SymIntNode pow(const SymIntNode& other) {
+    TORCH_CHECK(false, "NYI");
+  }
+  virtual SymFloatNode pow(const SymFloatNode& other) {
+    TORCH_CHECK(false, "NYI");
+  }
+  // TODO: should comparison sym ops return bool?
   virtual SymIntNode eq(const SymIntNode& other) {
     TORCH_CHECK(false, "NYI");
   };
+  virtual SymFloatNode eq(const SymFloatNode& other) {
+    TORCH_CHECK(false, "NYI");
+  };
   virtual SymIntNode ne(const SymIntNode& other) {
     TORCH_CHECK(false, "NYI");
   };
+  virtual SymFloatNode ne(const SymFloatNode& other) {
+    TORCH_CHECK(false, "NYI");
+  };
   virtual SymIntNode gt(const SymIntNode& other) {
     TORCH_CHECK(false, "NYI");
   };
+  virtual SymFloatNode gt(const SymFloatNode& other) {
+    TORCH_CHECK(false, "NYI");
+  };
   virtual SymIntNode lt(const SymIntNode& other) {
     TORCH_CHECK(false, "NYI");
   };
+  virtual SymFloatNode lt(const SymFloatNode& other) {
+    TORCH_CHECK(false, "NYI");
+  };
   virtual SymIntNode le(const SymIntNode& other) {
     TORCH_CHECK(false, "NYI");
   };
-  virtual SymIntNode ge(const SymIntNode& other) {
+  virtual SymFloatNode le(const SymFloatNode& other) {
```

Comment:
This looks wrong, less than or equals should return a bool (or failing that, an int)

Reply:
The current situation here is that both symfloat AND symint can contain bools. And both of them support `bool()` as expected.
I can move everything to say that only symint can contain a bool (and forbid calling `bool(symfloat)`.
I don't have any strong opinion either way.

@bdhirsh what's your vote?

Reply:
`SymBoolNode` feels like it would be nicer to have in the long term. Today, we have:

`SymIntNode` holds a pyobject, which we **guarantee** is always either a `PySymInt` or `int64_t (well, until nested tensor starts using them).
The `PySymInt` holds a `sympy.expr`, which... is always either an integer expression or a boolean expression

`SymFloatNode` holds a pyobject that's guaranteed to be either a `PySymFloat` or a `double`, and `PySymFloat` holds a `sympy.expr` which is always either a float expression or a boolean expression.

I guess for me, given that we already have a looser definition around what `PySymInt` can hold, it doesn't feel too bad for `PySymFloat` to act the same way? Unless we want to go all-in on the one-to-one correspondence and add `SymBool`.

I'm not strongly opinionated about this though.